### PR TITLE
infoschema: Migrate the infoschema's retrieving data logic for 'engine' to executor

### PR DESCRIPTION
**Pull request by AndrewDi**
### What problem does this PR solve? 
Infoschema contains too much logic for retrieving data, which is confusing, we need to migrate this part to the executor so that infoshema only contains schema information. Detail ref #15042 

### What is changed and how it works?
migrate the dataForEngine() from infoschema pkg to executor/infoschema_reader.go

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by AndrewDi** [comment]
@lonng PTAL

File: executor/builder.go [hunk 1]
```diff
@@ -1400,7 +1400,7 @@ func (b *executorBuilder) buildMemTable(v *plannercore.PhysicalMemTable) Executo
 					timeRange: v.QueryTimeRange,
 				},
 			}
-		case strings.ToLower(infoschema.TableSchemata), strings.ToLower(infoschema.TableViews):
+		case strings.ToLower(infoschema.TableSchemata), strings.ToLower(infoschema.TableViews), strings.ToLower(infoschema.TableEngines):
```
Comment (lonng) [inline executor/builder.go]: Prefer to split L1403 into multiple lines.

**Comment by reafans** [comment]
lgtm

**Pull request merged**